### DISCUSSION TITLE
Why can't I monitor file creation events?

### DISCUSSION THREAD

**Discussion by hooklab** [bpftrace/bpftrace#2275] [General]
```
kprobe:vfs_mkdir,
kprobe:vfs_create,
kprobe:vfs_mknod
{
 print("create");
}
```
when I  run  `mkdir  aadir `  , you can see the output ，  but run `touch aa.log` or  `vim bb.log`   or  `cp aa.log dd.log`,  no output 

**uanme -a** 
`Linux ying 5.18.5-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Jun 16 14:51:11 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by viktormalik** [comment]
The likely reason is that neither of those kernel functions are executed by `touch`. It looks like `touch` calls the `open` syscall with `O_CREAT` which means that the file creation is probably handled by `vfs_open`.

**Reply by hooklab** [reply]
> The likely reason is that neither of those kernel functions are executed by `touch`. It looks like `touch` calls the `open` syscall with `O_CREAT` which means that the file creation is probably handled by `vfs_open`.

Thank you, it is true, monitoring vfs_open has been successful.  And what command will be executed to go to vfs_create?
